wandb/sdk/wandb_run.py

```diff
@@ -44,6 +44,7 @@
     RunRecord,
```

Comment:
general question: does it make sense to have a Job(s) class and move all this business logic into that class?
I wonder if it would be cleaner that way

Reply:
there is a Job class that will be made available in the public API. But creating the jobs through the existing artifact infrastructure has some benefits like automatic de-duplication of versions if they are identical, and using an artifact browser to see the contents.

I suppose we could create a JobRecord and have the internal process handle it, but there'd be a lot of similar work that in my opinion is well handled by the log_artifact solution. Maybe @raubitsj has some thoughts here?

Reply:
hmmm... i was just thinking maybe some class `Job<something>` that would include all the logic that was added in the new functions like: `_has_job_reqs`, `_create_job`, `_create_repo_job`, `_create_artifact_job`, `_create_container_job` etc... basically moving it out of the `Run` class and only using this `Job` class to do all this logic? what do you think?

Reply:
We've decided to treat this more like log_code since the job is created through an artifact like code is. As such, `_create_job` will be renamed to `_log_job`.